packages/solid-urql/src/createQuery.ts

```diff
@@ -0,0 +1,337 @@
+ *
+ * @remarks
+ * the first element is the {@link createQuery}’s result and state,
+ * a {@link CreateQueryState} object,
+ * and the second is used to imperatively re-execute the query
+ * via a {@link CreateQueryExecute} function.
+ */
+export type CreateQueryResult<
+  Data = any,
+  Variables extends AnyVariables = AnyVariables,
+> = [CreateQueryState<Data, Variables>, CreateQueryExecute];
+
+/** Hook to run a GraphQL query and get updated GraphQL results.
+ *
+ * @param args - a {@link CreateQueryArgs} object, to pass a `query`, `variables`, and options.
+ * @returns a {@link CreateQueryResult} tuple of a {@link CreateQueryState} result, and re-execute function.
+ *
+ * @remarks
+ * `createQuery` allows GraphQL queries to be defined and executed.
+ * Given {@link CreateQueryArgs.query}, it executes the GraphQL query with the
+ * context’s {@link Client}.
+ *
+ * The returned result updates when the `Client` has new results
+ * for the query, and changes when your input `args` change.
+ *
+ * Additionally, if the `suspense` option is enabled on the `Client`,
+ * the `createQuery` hook will suspend instead of indicating that it’s
+ * waiting for a result via {@link CreateQueryState.fetching}.
+ *
+ * @example
+ * ```tsx
+ * import { gql, createQuery } from '@urql/solid';
+ *
+ * const TodosQuery = gql`
+ *   query { todos { id, title } }
+ * `;
+ *
+ * const Todos = () => {
+ *   const [result, reExecuteQuery] = createQuery({
+ *     query: TodosQuery,
+ *   });
+ *   // ...
+ * };
+ * ```
+ */
+export const createQuery = <
+  Data = any,
+  Variables extends AnyVariables = AnyVariables,
+>(
+  args: CreateQueryArgs<Data, Variables>
+): CreateQueryResult => {
+  const client = useClient();
+  const getContext = asAccessor(args.context);
+  const getPause = asAccessor(args.pause);
+  const getRequestPolicy = asAccessor(args.requestPolicy);
+  const getVariables = asAccessor(args.variables);
+
+  const [source, setSource] = createSignal<
+    Source<OperationResult<Data, Variables>> | undefined
+  >(undefined, { equals: false });
+
+  // Combine suspense param coming from context and client with context being priority
+  const isSuspense = createMemo(() => {
+    const ctx = getContext();
+    if (ctx !== undefined && ctx.suspense !== undefined) {
+      return ctx.suspense;
+    }
+
+    return client.suspense;
+  });
+
+  const request = createRequest(args.query, getVariables() as any);
+  const context: Partial<OperationContext> = {
+    requestPolicy: getRequestPolicy(),
+    ...getContext(),
+  };
+  const operation = client.createRequestOperation('query', request, context);
+  const initialResult: CreateQueryState<Data, Variables> = {
+    operation: operation,
+    fetching: false,
+    data: undefined,
+    error: undefined,
+    extensions: undefined,
+    hasNext: false,
+    stale: false,
+  };
+
+  const [result, setResult] =
+    createStore<CreateQueryState<Data, Variables>>(initialResult);
+
+  createComputed(() => {
```

Comment:
The implementation looks obviously quite similar between [`createComputed`](https://github.com/solidjs/solid/blob/f79ba4d9087d74ef7e55fd0dc336c98ce2ec8526/packages/solid/src/reactive/signal.ts#L282-L290) and [`createRenderEffect`](https://github.com/solidjs/solid/blob/f79ba4d9087d74ef7e55fd0dc336c98ce2ec8526/packages/solid/src/reactive/signal.ts#L313-L321).

The difference being a small `pure` flag that's being set.

However, I'm curious what the difference is because I find it a little odd to perform effects in what's essentially a utility that's supposed to be used to derive new signals

Reply:
Didn't delve deep into solid-js code base, but from my understanding
- `createComputed` - called immediataly on upadate, similar to `createMemo`
- `createRenderEffect` - behavior similar to `createEffect` difference is first execution primarly to be able to access refs during initial render